torch/_subclasses/fake_tensor.py

```diff
@@ -956,63 +959,271 @@ def extract_tensor_metadata(t: Tensor) -> TensorMetadata:
     Extract the TensorMetadata of a tensor.
     """
     memory_format: Optional[torch.memory_format] = suggest_memory_format(t)
-    if is_sparse_any(t) or not t.is_contiguous(memory_format=memory_format):
+    # Don't call is_contiguous() on a Tensor which has symbolic sizes or things
+    # will go badly (guards will be messed up?)
+    if (
+        t._has_symbolic_sizes_strides
+        or is_sparse_any(t)
+        or not t.is_contiguous(memory_format=memory_format)
+    ):
         memory_format = None
```

Comment:
Should we just remove memory_format from the key then ? Pytorch eager has a property on the tensor to distinguish ambiguous cases I believe and for propagating views of channels last, but it looks like in the symint case we just compute from strides.

Reply:
Trying it in #131052